pkg/infinity/headers.go

```diff
@@ -151,6 +153,18 @@ func ApplyForwardedOAuthIdentity(requestHeaders map[string]string, settings mode
 }
 
 func SetGrafanaUserHeader(query querySrv.Query, req *http.Request) *http.Request {
-	req.Header.Set(headerKeyGrafanaUser, query.GrafanaUser)
+	if query.GrafanaUser != "" {
+		req.Header.Set(headerKeyGrafanaUser, query.GrafanaUser)
+	} else {
+		// When no user in query, we assume the request is coming from alert backend parser
+		req.Header.Set(headerKeyGrafanaUser, "alert-backend")
```

Comment:
I don't honestly know under what circumstances the user wouldn't be set (apart from the alerting backend). It seems a bit presumptuous to say that it's only the alerting backend. Might be better to just set it to `null`, or not set it, etc.

Reply:
Yeah I meant the string `"null"` as a sort of placeholder, but not setting (or having it be blank) is fine too.